pnut.c

```diff
@@ -757,7 +763,35 @@ void output_declaration_c_code(bool no_header) {
 #endif
 
 void get_ch() {
```

Comment:
I think we could simplify the whole thing by separating `get_ch` in 2:
- One function (let's call it `get_ch_`) that calls `fgetc`, does the EOF handling and updates the location (like `get_ch`  from `main`).
- Another function (called `get_ch`) that calls `get_ch_` that loops when it encounters a `\\n` sequence.

That way we wouldn't need to repeat the location logic.

And when `SUPPORT_LINE_CONTINUATION` is off, `get_ch_` would simply be called `get_ch`.